packages/@n8n/decorators/src/types.ts

```diff
@@ -47,3 +47,14 @@ export interface ControllerMetadata {
 
 export type Controller = Constructable<object> &
 	Record<HandlerName, (...args: unknown[]) => Promise<unknown>>;
+
+type HandlerFn = () => Promise<void> | void;
```

Comment:
We should either move this to a routing specific file, or rename it to `RouteHandlerFn` instead.